test/blackbox-tests/test-cases/watching/what-dune-watches.t

```diff
@@ -21,6 +21,6 @@ The pattern below excludes absolute files as Dune currently watches
 everything in the PATH, which is not very reproducible.
 
   $ sed -nE 's/inotify_add_watch\([0-9]*, "([^/].*)", .*\) (=.*)/watch \1 \2/p' ../log | sort -u
-  watch . = 2
+  watch . = 1
   watch _build/.sync = 1
-  watch src = 3
+  watch src = 2
```

Comment:
do we know why this test  changed?

Reply:
(This was super painful; I had to setup Linux to test this. It of course doesn't work on Windows, but it doesn't work on macOS either).

I updated the code to include the full set of watched directories. For whatever reason Linux is re-using the watch number (the return code of `inotify_add_watch`) for `.` and `_build/.sync` in the new code, while the original code has them separate.

The original code resulted in:

```diff
+|  watch _build/.sync = 1
+|  watch . = 2
+|  watch src = 3
+|  watch DUNE_SOURCEROOT/_build/default/test/blackbox-tests/test-cases/.bin = 4
+|  watch DUNE_SOURCEROOT/_build/install/default/bin = 5
+|  watch OPAM_SWITCH_PREFIX/bin = 6
+|  watch OPAM_SWITCH_PREFIX/lib/ocaml = 7
```

Reply:
Fixed. Was initializing the inotify watcher twice during the switch from singleton to a parameterized function. (Last commit).